File: src/main/java/com/itsci/mju/maebanjumpen/repository/TransactionRepository.java
package com.itsci.mju.maebanjumpen.repository;

import com.itsci.mju.maebanjumpen.model.Transaction;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.EntityGraph;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface TransactionRepository extends JpaRepository<Transaction, Integer> {

    List<Transaction> findByTransactionTypeAndTransactionStatus(String transactionType, String transactionStatus);


    List<Transaction> findByMemberId(int memberId);


    @Override
    @EntityGraph(attributePaths = {"member", "member.person", "member.person.login"})
    Optional<Transaction> findById(Integer id);

    List<Transaction> findByTransactionType(String transactionType);
    List<Transaction> findByTransactionTypeOrTransactionType(String type1, String type2);
}

File: src/main/kotlin/com/itsci/mju/maebanjumpen/transaction/repository/TransactionRepository.kt
package com.itsci.mju.maebanjumpen.transaction.repository

import com.itsci.mju.maebanjumpen.entity.Transaction
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface TransactionRepository : JpaRepository<Transaction, Long> {

    fun findByTransactionTypeAndTransactionStatus(transactionType: String, transactionStatus: String): List<Transaction>

    fun findByMemberId(memberId: Int): List<Transaction>

    @EntityGraph(attributePaths = ["member", "member.person", "member.person.login"])
    override fun findById(id: Long): Optional<Transaction>

    fun findByTransactionType(transactionType: String): List<Transaction>

    fun findByTransactionTypeOrTransactionType(type1: String, type2: String): List<Transaction>
}

